core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1343,7 +1343,8 @@ private void substituteSubQueryInSomeAll(Blackboard bb, SubQuery subQuery) {
         for (int i = 0; i < leftKeys.size(); i++) {
           RexNode lhsNode = leftKeys.get(i);
           RexNode rhsNode = this.rexBuilder.makeInputRef(
-              rhsFields.getType().getFieldList().get(i).getType(), outputStartingIdx);
+              rhsFields.getType().getFieldList().get(i).getType(),
+              outputStartingIdx + i);
```

Comment:
To clarify this is because there are potentially multiple keys for the same starting idx?

Reply:
I want to make sure this is explained to me before I approve.

Reply:
So, in the case that we have some expression like "(A, B) in (X, Y)",

The leftkeys will be A and B, and the root of the blackboard will be (outputStartingIdx number of rows), X, Y.

Originally, what was happening here was that we weren't incrementing the inputRef that was being generated for the right hand side of the equality. IE, the condition ended up being:

A == X AND B == X

With this change, we should generate:

A == X and B == Y

Reply:
Okay this matches my expectations. Thank you!